### ISSUE TITLE
Client starts with blank grey screen on some macOS computers

### ISSUE THREAD

**Issue by technolion** [flutter/flutter#49426]
## Steps to Reproduce

We built a Flutter based macOS app, that runs well on most macOS computers. On some however it does not start up properly. It looks like this:

![mac-client-does-not-start](https://user-images.githubusercontent.com/1611505/73077217-0c58a380-3ec0-11ea-8ba3-d05326d15983.png)

The Console shows the following log output:

`MessageTracer: load_domain_whitelist_search_tree:73: Search tree file's format version number (0) is not supported`

The last computer who showed the problem was running Mac OS X 10.13.6 High Sierra, but we have also seen this problem on macOS 10.15.2. All of the computers, where the problem arises no flutter build chain is installed. Those were external tester machines.

I am happy to provide the built Application for test reasons.

This looks a little bit like this one year old bug:
https://github.com/google/flutter-desktop-embedding/issues/245

<details>
  <summary>Logs</summary>

porcupine:skp-client tom$ flutter analyze
Analyzing skp-client...                                                 

   info • Unused import: 'package:intl/intl.dart' • lib/dialogs.dart:7:8 • unused_import
   info • The value of the local variable 'l10n' isn't used • lib/main.dart:110:8 • unused_local_variable
   info • The value of the local variable 'response' isn't used • lib/transferDetails.dart:634:11 • unused_local_variable

3 issues found. (ran in 2.2s)

```
```

flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel master, v1.14.4-pre.45, on Mac OS X 10.15.2 19C57, locale de-DE)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
[!] Android Studio (version 3.4)
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
[✓] VS Code (version 1.41.1)
[✓] Connected device (2 available)

! Doctor found issues in 1 category.

```
```

</details>

**Comment by technolion** [comment]
@stuartmorgan please take a look.

**Comment by VladyslavBondarenko** [comment]
Hi @technolion 
I would advice you better create the issue in https://github.com/google/flutter-desktop-embedding and delete current one since this isn't an issue with Flutter itself

**Comment by technolion** [comment]
@VladyslavBondarenko
the macOS platform has become an official part of the main flutter project (on the master channel) some time ago. There is no need to use flutter-desktop-embedding any longer for macOS builds.
In fact I created the app by simply running **flutter create**.

**Comment by VladyslavBondarenko** [comment]
@technolion sorry)

**Comment by stuartmorgan** [comment]
> This looks a little bit like this one year old bug:
> google/flutter-desktop-embedding#245

Any bug that causes nothing to ever draw will look like that bug.

Have you tried running a debug build on one of the failing machines to see what is logged?

**Comment by technolion** [comment]
I finally got the chance to run a debug build of a machine, where the blank screen appeared. 
And it turned out that it throwed a `bad state: no element` error shown on a red screen.  Together with the stack information I found out, that it was triggered by a `firstWhere()` method on an `Iterable`.
I had to provide a orElse to prevent that situation. More info [here](https://stackoverflow.com/questions/52354195/list-firstwhere-bad-state-no-element/52354226).
Since this happened in flutter's main.dart the initial screen was grey (only on release builds).

So this is not a Flutter bug, sorry about the fuzz. This can be closed.
